### DISCUSSION TITLE
how to attach drivers to devices in WSL2

### DISCUSSION THREAD

**Discussion by SCIEN-00** [microsoft/WSL#9023] [Q&A]
Hei!

I try to connect my xbox controller to WSL2 (need to use it in ROS).
I got xbox controller attached to wsl2 using usbipd and it appears under /dev/bus/usb/00x/ devices but no drivers "attached".
![image](https://user-images.githubusercontent.com/11405851/196004518-1fa1866d-fb50-4979-8614-7c177f98d678.png)
Also, it did not appear under /dev/input/jsx device.
Under Linux computer it looks like this
![image](https://user-images.githubusercontent.com/11405851/196004573-2d9f955a-84c7-454a-aa21-adb9f672f100.png)
I see that there is xpad driver in wsl kernel https://github.com/microsoft/WSL2-Linux-Kernel/blob/master/drivers/input/joystick/xpad.c. How should I make that device under dev/bus/usb/00x to use this xpad driver?

**Comment by KevinDW007** [comment]
Did you ever find a solution to this? im trying to do the same thing
